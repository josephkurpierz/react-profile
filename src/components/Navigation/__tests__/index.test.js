import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

const categories = [
  { name: "Projects" },
];
const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();
const mockContactSelected = jest.fn();
const mockSetContactSelected = jest.fn();

afterEach(cleanup);

describe("Nav component", () => {
  //baseline test
  it("renders", () => {
    render(
      <Nav
        categories={categories}
        setCurrentCategory={mockSetCurrentCategory}
        currentCategory={mockCurrentCategory}
        contactSelected={mockContactSelected}
        setContactSelected={mockSetContactSelected}
      />
    );
  });
  // snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Nav
        categories={categories}
        setCurrentCategory={mockSetCurrentCategory}
        currentCategory={mockCurrentCategory}
        contactSelected={mockContactSelected}
        setContactSelected={mockSetContactSelected}
      />
    );
    //assert value comparison
    expect(asFragment()).toMatchSnapshot();
  });
});

// describe("emoki is visible", () => {
//   it("inserts emoji into the h2", () => {
//     //arrange
//     render(
//       <Nav
//         categories={categories}
//         setCurrentCategory={mockSetCurrentCategory}
//         currentCategory={mockCurrentCategory}
//         contactSelected={mockContactSelected}
//         setContactSelected={mockSetContactSelected}
//       />
//     );
//     //assert
//     expect(screen.getByLabelText("camera")).toHaveTextContent("ğŸ“¸");
//   });
// });

// u
