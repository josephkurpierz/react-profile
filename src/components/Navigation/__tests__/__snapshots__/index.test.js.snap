// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      id="nav"
    >
      <p>
        this is text in the nav section
      </p>
      <button
        class="false"
      >
        <span>
          Projects
        </span>
      </button>
    </nav>
  </div>
</DocumentFragment>
`;
